src/test/java/org/opensearch/neuralsearch/query/aggregation/BucketAggregationsWithHybridQueryIT.java

```diff
@@ -0,0 +1,925 @@
+    private static final String SEARCH_PIPELINE = "search-pipeline-bucket-aggs";
+
+    @SneakyThrows
+    public void testBucketAndNestedAggs_whenAdjacencyMatrix_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testAdjacencyMatrixAggs();
+    }
+
+    @SneakyThrows
+    public void testWithConcurrentSegmentSearch_whenAdjacencyMatrix_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
+        testAdjacencyMatrixAggs();
+    }
+
+    @SneakyThrows
+    public void testEmptyResults_whenHybridQueryResultOrAggregationsEmptyResults_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testSumNestedIntoFiltersWithEmptyResults();
+    }
+
+    @SneakyThrows
+    public void testEmptyResultsWithConcurrentSegmentSearch_whenHybridQueryResultOrAggregationsEmptyResults_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
+        testSumNestedIntoFiltersWithEmptyResults();
+    }
+
+    @SneakyThrows
+    public void testBucketAndNestedAggs_whenDiversifiedSampler_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testDiversifiedSampler();
+    }
+
+    @SneakyThrows
+    public void testWithConcurrentSegmentSearch_whenDiversifiedSampler_thenFail() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
+
+        testDiversifiedSampler();
+    }
+
+    @SneakyThrows
+    public void testBucketAndNestedAggs_whenAvgNestedIntoFilter_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testAvgNestedIntoFilter();
+    }
+
+    @SneakyThrows
+    public void testWithConcurrentSegmentSearch_whenAvgNestedIntoFilter_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
+        testAvgNestedIntoFilter();
+    }
+
+    @SneakyThrows
+    public void testBucketAndNestedAggs_whenSumNestedIntoFilters_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testSumNestedIntoFilters();
+    }
+
+    @SneakyThrows
+    public void testWithConcurrentSegmentSearch_whenSumNestedIntoFilters_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
+        testSumNestedIntoFilters();
+    }
+
+    @SneakyThrows
+    public void testBucketAggs_whenGlobalAggUsedWithQuery_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, false);
+        testGlobalAggs();
+    }
+
+    @SneakyThrows
+    public void testWithConcurrentSegmentSearch_whenGlobalAggUsedWithQuery_thenSuccessful() {
+        updateClusterSettings(CLUSTER_SETTING_CONCURRENT_SEGMENT_SEARCH, true);
```

Comment:
nit: can we refactor to two class with concurrent segment and other without? You can enable at class set up once and don't have to flip in every test.

Reply:
this may not work as two different classes may be picked up by  different test runners and execute in parallel. In such case exact value of setting is not guaranteed.

Reply:
Can we use index setting to enable/disable concurrent segment search then?

Reply:
Please create a github issue for refactor and link this comment.

Reply:
Sure, created a new GH issue https://github.com/opensearch-project/neural-search/issues/699